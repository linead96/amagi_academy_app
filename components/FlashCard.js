import React from 'react';
import {
  Text,
  StyleSheet,
  Animated,
  Easing,
} from 'react-native';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
  flashcardQuestion: {
    flex: 6,
    justifyContent: 'center',
    alignItems: 'center',
    width: '80%',
    height: '60%',
    borderRadius: 30,
    marginTop: '10%',
    marginBottom: 10,
  },
  flashcardAnswer: {
    flex: 6,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#11CACA',
    width: '80%',
    height: '60%',
    borderRadius: 30,
    marginTop: '10%',
    marginBottom: 10,
  },
  flashcardQuestionText: {
    textAlign: 'center',
    fontSize: 30,
    margin: '10%',
    color: 'white',
  },
  flashcardAnswerText: {
    textAlign: 'center',
    fontSize: 30,
    margin: '10%',
    color: 'white',
  },
});

class FlashCard extends React.Component {
  constructor(props) {
    super(props);
    this.animatedValue = new Animated.Value(0);
  }

  componentDidUpdate(prevProps) {
    const { flipped } = this.props;
    if (flipped !== prevProps.flipped) {
      this.animate();
    }
  }

  animate() {
    this.animatedValue.setValue(0);
    Animated.timing(
      this.animatedValue,
      {
        toValue: 1,
        duration: 250,
        easing: Easing.linear,
      },
    ).start();
  }

  stopAnimate() {
    Animated.timing(this.animatedValue, {
      toValue: 0,
      duration: 0,
      easing: Easing.linear,
      useNativeDriver: true,
    }).start();
  }


  render() {
    const {
      flipped,
      bg,
      question,
      answer,
    } = this.props;

    const rotateX = this.animatedValue.interpolate({
      inputRange: [0, 0.99, 1],
      outputRange: ['0deg', '180deg', '0deg'],
    });

    return flipped
      ? (
        <Animated.View style={{
          ...styles.flashcardAnswer,
          backgroundColor: bg,
          transform: [{ rotateX }],
        }}
        >
          <Text style={styles.flashcardAnswerText}>
            {`Answer:\n${answer}`}
          </Text>
        </Animated.View>
      )
      : (
        <Animated.View style={{
          ...styles.flashcardQuestion,
          backgroundColor: bg,
          transform: [{ rotateX }],
        }}
        >
          <Text style={styles.flashcardQuestionText}>
            {`Question:\n${question}`}
          </Text>
        </Animated.View>
      );
  }
}

FlashCard.propTypes = {
  flipped: PropTypes.bool.isRequired,
  bg: PropTypes.string.isRequired,
  question: PropTypes.string.isRequired,
  answer: PropTypes.string.isRequired,
};

export default FlashCard;
