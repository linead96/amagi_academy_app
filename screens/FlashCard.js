import React from 'react';
import {
  StyleSheet,
  View,
  Text
} from 'react-native';

export default class FlashCard extends React.Component {
  static navigationOptions = {

  };

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.flashcard}>
            <Text style={styles.text}>Ut porttitor lectus vehicula, mattis nulla vel, commodo nisi.</Text>
        </View>
        <View style={styles.flipButton}>
            <Text style={styles.flipButton__text}>Flip</Text>
        </View>
      </View>
    );
  }

}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    fontSize: 20,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#fff',
  },
  flashcard: {
    flex: 6,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#90F145', 
    width: '80%',
    height: '60%',
    borderRadius: 10,
    marginTop: '20%',    
    marginBottom: 10,
  },
  text: {
    textAlign: 'center',
    fontSize: 30,
    margin: '10%',
  },   
  flipButton: {
    backgroundColor: 'yellow',  
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 25,
    marginBottom: '5%',
  },
  flipButton__text: {
    textAlign: 'center',
    fontSize: 30,
    margin: '10%',
  },
});
