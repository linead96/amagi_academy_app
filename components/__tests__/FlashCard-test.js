import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import FlashCard from '../FlashCard';


describe('test flashcardComponent', () => {
  test('renders a view with a question when the state flipped is false', () => {
    const tree = renderer.create(
      <FlashCard
        bg="#FFFFFF"
        flipped={false}
        question="Lorem ipsum dolor sit amet, consectetur adipiscing elit."
        answer="Lorem ipsum dolor sit amet."
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders a view with an answer when the state flipped is true', () => {
    const tree = renderer.create(
      <FlashCard
        bg="#FFFFFF"
        flipped
        question="Lorem ipsum dolor sit amet, consectetur adipiscing elit."
        answer="Lorem ipsum dolor sit amet."
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
