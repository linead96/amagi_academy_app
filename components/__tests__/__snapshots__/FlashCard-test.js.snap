// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test flashcardComponent renders a view with a question when the state flipped is false 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 30,
      "flex": 6,
      "height": "60%",
      "justifyContent": "center",
      "marginBottom": 10,
      "marginTop": "10%",
      "transform": Array [
        Object {
          "rotateX": "0deg",
        },
      ],
      "width": "80%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 30,
        "margin": "10%",
        "textAlign": "center",
      }
    }
  >
    Question:
Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  </Text>
</View>
`;

exports[`test flashcardComponent renders a view with an answer when the state flipped is true 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 30,
      "flex": 6,
      "height": "60%",
      "justifyContent": "center",
      "marginBottom": 10,
      "marginTop": "10%",
      "transform": Array [
        Object {
          "rotateX": "0deg",
        },
      ],
      "width": "80%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 30,
        "margin": "10%",
        "textAlign": "center",
      }
    }
  >
    Answer:
Lorem ipsum dolor sit amet.
  </Text>
</View>
`;
